tools/crm_mon.c

```diff
@@ -4868,11 +4932,15 @@ clean_up(int rc)
     }
 
     free(output_filename);
-    free(xml_file);
     free(pid_file);
 
-    if (rc >= 0) {
-        crm_exit(rc);
+    if (exit_code == EX_USAGE) {
+        if (output_format == mon_output_cgi) {
+            fprintf(stdout, "Content-Type: text/plain\n"
+                            "Status: 500\n\n");
+        } else {
+            crm_help('?', EX_USAGE);
+        }
     }
-    return;
+    crm_exit(exit_code);
```

Comment:
I think we need to keep the "if (rc >= 0)" around this to preserve the old behavior. I believe the only possible negative values are if the CIB connection/setup fails or the stonith connection fails while already connected to the CIB. I don't remember why I changed those to fatal errors for 2.0, but we probably don't want to change it in 1.1.

Reply:
Till the first connection probably anything but -ENOTCON should be fatal.
And afterwards nothing should be fatal because we can always hope for the state returning to what we hab when it worked for the first time.
Why then don't we split it up into void clean_up_connections(void) we would call from cib_connect instead of clean_up and from refactored clean_up that would be used - as is - everywhere else.

Reply:
OK, I'll merge this as-is and then we can do that in master + backport later.